ecommerce/OrderController.php

```diff
@@ -156,6 +163,48 @@ public function get_order_by_id() {
 		);
 	}
 
+	/**
+	 * Mark an order as paid.
+	 *
+	 * This function verifies a nonce for security, constructs a payload object with
+	 * the order ID, note, and payment status, and updates the payment status of the order
+	 * to 'paid'. It sends a JSON response indicating the success or failure of the operation.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function order_mark_as_paid() {
+		if ( ! tutor_utils()->is_nonce_verified() ) {
+			$this->json_response( tutor_utils()->error_message( 'nonce' ), null, HttpHelper::STATUS_BAD_REQUEST );
+		}
+
+		$payload                 = new \stdClass();
+		$payload->order_id       = Input::post( 'order_id' );
+		$payload->note           = Input::post( 'note' );
+		$payload->payment_status = $this->model::PAYMENT_STATUS_PAID;
+
+		if ( empty( $payload->order_id ) ) {
```

Comment:
Use validation rules & validation helper for validating inputs. Check `CourseController::validate` method from `tutor-pro` : **app/public/wp-content/plugins/tutor-pro/rest-api/Controllers/CourseController.php**